test/controllers/AssociatedCompaniesControllerSpec.scala

```diff
@@ -0,0 +1,168 @@
+
+        val result = route(application, request).value
+
+        status(result) mustEqual OK
+        contentAsString(result) mustEqual view(
+          form.fill(AssociatedCompaniesForm(AssociatedCompanies.Yes, Some(1))),
+          NormalMode
+        )(request, messages(application)).toString
+      }
+    }
+
+    "must redirect to the next page when valid data is submitted" in {
+
+      val mockSessionRepository = mock[SessionRepository]
+
+      when(mockSessionRepository.set(any())) thenReturn Future.successful(true)
+
+      val application =
+        applicationBuilder(userAnswers = Some(emptyUserAnswers))
+          .overrides(
+            bind[Navigator].toInstance(new FakeNavigator(onwardRoute)),
+            bind[SessionRepository].toInstance(mockSessionRepository)
+          )
+          .build()
+
+      running(application) {
+        val request =
+          FakeRequest(POST, associatedCompaniesRoute)
+            .withFormUrlEncodedBody(("associatedCompanies", "yes"), ("associatedCompaniesCount", "1"))
+
+        val result = route(application, request).value
+
+        status(result) mustEqual SEE_OTHER
+        redirectLocation(result).value mustEqual onwardRoute.url
+      }
+    }
+
+    "must return a Bad Request and errors when invalid data is submitted" in {
+
+      val application = applicationBuilder(userAnswers = Some(emptyUserAnswers)).build()
+
+      running(application) {
+        val request =
+          FakeRequest(POST, associatedCompaniesRoute)
+            .withFormUrlEncodedBody(
+              ("associatedCompanies", "invalid value"),
+              ("associatedCompaniesCount", "invalid value")
+            )
+
+        val boundForm =
+          form.bind(Map("associatedCompanies" -> "invalid value", "associatedCompaniesCount" -> "invalid value"))
+
+        val view = application.injector.instanceOf[AssociatedCompaniesView]
+
+        val result = route(application, request).value
+
+        status(result) mustEqual BAD_REQUEST
+        contentAsString(result) mustEqual view(boundForm, NormalMode)(request, messages(application)).toString
+      }
+    }
+
+    "must redirect to Journey Recovery for a GET if no existing data is found" in {
+
+      val application = applicationBuilder(userAnswers = None).build()
+
+      running(application) {
+        val request = FakeRequest(GET, associatedCompaniesRoute)
+
+        val result = route(application, request).value
+
+        status(result) mustEqual SEE_OTHER
+        redirectLocation(result).value mustEqual routes.JourneyRecoveryController.onPageLoad().url
+      }
+    }
+
+    "redirect to Journey Recovery for a POST if no existing data is found" in {
+
+      val application = applicationBuilder(userAnswers = None).build()
+
+      running(application) {
+        val request =
+          FakeRequest(POST, associatedCompaniesRoute)
+            .withFormUrlEncodedBody(("value", AssociatedCompanies.values.head.toString))
```

Comment:
are you sure this is right? You seem to be sending wrong parameters i.e value isnt supported

Reply:
fixed